source/cellular_r4_api.c

```diff
@@ -2495,22 +2495,43 @@ static CellularPktStatus_t _Cellular_RecvFuncGetPdpContextSettings( CellularCont
                             case ( CELLULAR_PDN_STATUS_POS_CONTEXT_TYPE ):
                                 LogDebug( ( "_Cellular_RecvFuncGetPdpContextSettings: Context Type pToken: %s", pToken ) );
 
-                                ( void ) memcpy( ( void * ) pPDPContextsInfo->ipType[ contextId - 1 ],
-                                                 ( void * ) pToken, CELULAR_PDN_CONTEXT_TYPE_MAX_SIZE + 1U );
+                                ( void ) strncpy( ( void * ) pPDPContextsInfo->ipType[ contextId - 1 ],
+                                                  ( void * ) pToken, CELULAR_PDN_CONTEXT_TYPE_MAX_SIZE + 1U );
+
+                                if( pPDPContextsInfo->ipType[ contextId - 1 ][ CELULAR_PDN_CONTEXT_TYPE_MAX_SIZE ] != '\0' )

```

Comment:
Do we initialize the buffer to all 0? Or how do we gaurantee that latest char must be terminator?

Reply:
strncpy will padded the destination with zeros until a total of num characters have been written to it. If the source is longer than destination, no null-character will be appended.